### ISSUE TITLE
`appenv-python-preference` does not work with `update-lockfile`

### ISSUE THREAD

**Issue by sallner** [flyingcircusio/batou#187]
When I have a python-preference in the `requirements.txt` 
```
# appenv-python-preference: 3.6
...
``` 
then the subsequent call of `./appenv update-lockfile` fails with the following error:

```
$./appenv update-lockfile
Updating lockfile
Creating venv ...
Ensuring pip ...
Installing packages ...
Traceback (most recent call last):
  File "./appenv", line 447, in <module>
    main()
  File "./appenv", line 439, in main
    appenv.meta()
  File "./appenv", line 232, in meta
    args.func(args, remaining)
  File "./appenv", line 401, in update_lockfile
    spec = list(pkg_resources.parse_requirements(line))[0]
IndexError: list index out of range
```
This is rooted in this code section, where comments are not excluded during parsing of `requirements.txt` https://github.com/flyingcircusio/batou/blob/638038744281c804f2ad02c95c141f48b476b8fd/appenv.py#L398-L400

System:

Batou 2.3b1
Python 3.6

**Comment by jugmac00** [comment]
fun fact - `startswith` can not only take a string, but also a tuple (and no, I have not looked at this problem in particular)

https://docs.python.org/3/library/stdtypes.html#str.startswith

**Comment by sallner** [comment]
I added a workaround for it locally by skipping lines starting with `#` but plan to make a PR with a proper test, if possible.

**Comment by sallner** [comment]
This issue was solved with #207 so I am closing it.
